core/src/main/scala/org/ensime/core/javac/JavaCompiler.scala

```diff
@@ -0,0 +1,162 @@
+package org.ensime.core.javac
+
+import akka.event.slf4j.SLF4JLogging
+import com.sun.source.tree.Scope
+import com.sun.source.util.{ JavacTask, TreePath, Trees }
+import java.io.{ File, FileInputStream, InputStream }
+import java.net.URI
+import java.nio.charset.Charset
+import java.util.Locale
+import javax.lang.model.`type`.TypeMirror
+import javax.tools._
+import org.ensime.api._
+import org.ensime.core.DocSigPair
+import org.ensime.util.ReportHandler
+import org.ensime.util.file._
+import scala.collection.JavaConverters._
+import scala.reflect.internal.util.{ BatchSourceFile, RangePosition, SourceFile }
+import scala.tools.nsc.Settings
+import scala.tools.nsc.interactive.CompilerControl
+import scala.tools.nsc.io.AbstractFile
+import scala.tools.nsc.reporters.Reporter
+import scala.tools.refactoring.analysis.GlobalIndexes
+
+class JavaCompiler(
+    val config: EnsimeConfig,
+    val reportHandler: ReportHandler
+) extends JavaDocFinding with JavaCompletion with SLF4JLogging {
+
+  private val listener = new JavaDiagnosticListener()
+  private val silencer = new SilencedDiagnosticListener()
+
+  // needs to be recreated in JDK6. JDK7 seems more capable of reuse.
+  def getTask(
+    lint: String,
+    listener: DiagnosticListener[JavaFileObject],
+    files: List[JavaFileObject]
+  ): JavacTask = {
+    // TODO: take a charset for each invocation
+    val compiler = ToolProvider.getSystemJavaCompiler()
+    val fileManager = compiler.getStandardFileManager(listener, null, DefaultCharset)
```

Comment:
actually, if you're benchmarking, can you also please check that the various javac tasks and file managers are correctly garbage collected. Thought just occured to me after merging.